Dikubot/Webapp/Shared/Facebook/FacebookAddPages.razor

```diff
@@ -0,0 +1,143 @@
+            </MudCard>
+        </ChildRowContent>
+    </MudDataGrid>
+
+}
+
+
+
+
+@code {
+    [Inject] private HttpClient HttpClient { get; set; }
+    
+    private List<FBPage> _fbPagesList = new List<FBPage>();
+    
+    private void ConnectFacebookAction()
+    {
+        string link =
+            "https://www.facebook.com/v16.0/dialog/oauth?" +
+            "client_id=" + Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID") +
+            "&redirect_uri="+_navigationManager.BaseUri +"/FacebookAddPages" +
+            "&auth_type=rerequest" +
+            "&config_id=1372246313628981";
+
+        _navigationManager.NavigateTo(link);
+    }
+    
+
+    protected override void OnAfterRender(bool firstRender)
+    {
+        if (_navigationManager.Uri.Contains("code") && firstRender)
+        {
+            var appId = Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID"); 
+            var redirectUri = _navigationManager.BaseUri + "FacebookAddPages";
+            var appSecret = Environment.GetEnvironmentVariable("FACEBOOK_APP_SECRET"); 
+            var graphApiVersion = "v16.0";
+            try
+            {
+                string header = _navigationManager.Uri.Remove(0, _navigationManager.Uri.IndexOf('?') + 1);
+
+                
+                
+                var codeParameter = header.Remove(0, 5);
+
+    // Link to get the short lived User AccessToken
+                var shortlivedAccessTokenLink = $"https://graph.facebook.com/{graphApiVersion}/oauth/access_token?client_id={appId}&redirect_uri={redirectUri}&client_secret={appSecret}&code={codeParameter}";
+
+                FBAccessToken shortLivedAccessToken = HttpClient.GetFromJsonAsync<FBAccessToken>(shortlivedAccessTokenLink).Result;
+                
+                // Link to get the short lived User AccessToken
+                var getFacebookUsersPagesLink = $"https://graph.facebook.com/{graphApiVersion}/me/accounts?access_token={shortLivedAccessToken.AccessToken}";
+
+                _fbPagesList = HttpClient.GetFromJsonAsync<FBPageRespons>(getFacebookUsersPagesLink).Result.Data;
+
+                // Link to get the Long lived User AccessToken
+                var longLivedAccessTokenLink = $"https://graph.facebook.com/{graphApiVersion}/oauth/access_token?grant_type=fb_exchange_token&" +
+                            $"client_id={appId}&" +
+                            $"client_secret={appSecret}&" +
+                            $"fb_exchange_token={shortLivedAccessToken.AccessToken}";
+                var longLivedUserAccessToken = HttpClient.GetFromJsonAsync<FBAccessToken>(longLivedAccessTokenLink).Result;
+                
+                    
+                // Link to get the unique app User id
+                var userIdLink = $"https://graph.facebook.com/me?fields=id&access_token={longLivedUserAccessToken.AccessToken}";
+                var appScopedUserId = HttpClient.GetFromJsonAsync<FBidRespons>(userIdLink).Result.Id;
+
```

Comment:
It seems like a lot of this could be added to the aforementioned FacebookService.cs